sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +189,472 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
 	h.connectionsSequenceID = sequenceID
 
 	err := h.Process(networkInfoOpen, nowTimestamp, sequenceID)
-	suite.NoError(err)
+	s.NoError(err)
+
+	s.Len(h.endpoints, 1)
+}
+
+// endregion
+
+// region networkFlowManager tests
+
+func (s *NetworkFlowManagerTestSuite) TestEnrichConnection() {
+	mockCtrl := gomock.NewController(s.T())
+	m, mockEntityStore, mockExternalSrc, _ := createManager(mockCtrl)
+	srcID := "src-id"
+	dstID := "dst-id"
+	cases := map[string]struct {
+		connPair                    *connectionPair
+		enrichedConnections         map[networkConnIndicator]timestamp.MicroTS
+		expectEntityLookupContainer expectFn
+		expectEntityLookupEndpoint  expectFn
+		expectExternalLookup        expectFn
+		expectedIndicator           *networkConnIndicator
+		expectedConnection          *connection
+		expectedStatus              *connStatus
+	}{
+		"Rotten connection should return rotten status": {
+			connPair: createConnectionPair().incoming().external().firstSeen(timestamp.Now().Add(-maxContainerResolutionWaitPeriod * 2)),
+			expectEntityLookupContainer: func() {
+				mockEntityStore.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
```

Comment:
This expectation has the same parameters and times called in every test case. Let's not define this for each test case and simplify it a bit. 

Why not have a `clusterentities.ContainerMetadata` property in the test case, called `givenContainerMetadata`. This way each test case would define its own, and for the empty case you just don't define it (as the empty value for a struct is the struct itself with all empty values).